mlflow/xgboost.py

```diff
@@ -255,19 +256,43 @@ def log_model(
 
 
 def _load_model(path):
-    import xgboost as xgb
+    """
+    Load Model Implementation.
 
-    model = xgb.Booster()
-    model.load_model(os.path.abspath(path))
+    :param path: Local filesystem path to
+                    the MLflow Model with the ``xgboost`` flavor (MLflow < x.x.x) or
+                    the top-level MLflow Model directory (MLflow >= x.x.x).
+    """
+    import importlib
+
+    model_dir = os.path.dirname(path) if os.path.isfile(path) else path
+    flavor_conf = _get_flavor_configuration(model_path=model_dir, flavor_name=FLAVOR_NAME)
+
+    # XGBoost Booster models saved in MLflow (<x.x.x) specify
+    # the ``data`` field within its pyfunc and XGBoost flavor configurations.
+    # In contrast, XGBoost models saved in new MLflow (>=x.x.x) do not
+    # specify the ``data`` field within its pyfunc flavor configuration.
+    # We also add ``model_class`` in XGBoost flavor configuration to specify
+    # its XGBoost model class. When loading models, we first get the XGBoost
+    # model from its flavor configuration and then create an instance based on its class.
```

Comment:
Can we fix this comment because we'll keep the data field?

Reply:
The `data`field in `pyfunc` flavor was removed [(L157)](https://github.com/mlflow/mlflow/pull/4954/files#diff-f6bd607049d4969297151a812f447bdc13453a856df017983f7ea81470d25bd0R157) to avoid extra work in `_load_pyfunc()` ([discussion](https://github.com/mlflow/mlflow/pull/4954#discussion_r742956137)). We can remove the comment if the change is clear. Or add back the `data` field to the `pyfunc` flavor. Please let me know which is preferred. Thanks!

Reply:
Sorry I misread the code diff. `pyfunc` flavor no longer has the `data` field. I'd prefer to keep the `data` field since removing it might lead to a breaking change.

@dbczumar Any thoughts?

Reply:
@harupy What's the backwards-incompatible change here? Is the issue that older versions of MLflow won't be able to load XGBoost Booster models saved after this change is applied? If so, I don't think we typically guarantee that new MLflow models are loadable by older MLflow versions.

If we want to maintain this guarantee by preserving the `data` field, how do we propose to find the model class within `_load_pyfunc`?

Reply:
> If we want to maintain this guarantee by preserving the `data` field, how do we propose to find the model class within `_load_pyfunc`?

We can still get the directory path by retrieving the parent directory of `model.xgb` first as done in [L268](https://github.com/mlflow/mlflow/pull/4954/files#diff-f6bd607049d4969297151a812f447bdc13453a856df017983f7ea81470d25bd0R268).

Reply:
@jwyyy Sounds good! I think it's okay to assume that `model.xgb` always lives within the same directory as `MLmodel`, since this flavor is in charge of determining the directory structure for the model.

Reply:
@dbczumar Thank you for your suggestion! @harupy I think it is settled now. I will add back `data` in the `pyfunc` flavor after the current check finishes. (It failed Keras 2.7.0 autologging / models, and the cause seems to be the compatibility issue that happened in other PRs a few days ago. I guess it may also fail TF 2.7.0 tests as well.)

Reply:
Sounds good, thanks for the updates!

Reply:
We can ignore keras 2.7.0 failures since we solved them in #5024.